detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmYamlTransformerv6.java

```diff
@@ -0,0 +1,201 @@
+                        addTransitiveDependencyToGraph(graphBuilder, linkedPackageResolver, reportingProjectPackagePath,
+                                pnpmPackage, packageDependency);
+                    }
+                }
+            }
+        }
+    }
+
+    private void addTransitiveDependencyToGraph(DependencyGraph graphBuilder,
+            PnpmLinkedPackageResolver linkedPackageResolver, String reportingProjectPackagePath,
+            Optional<Dependency> pnpmPackage, Map.Entry<String, String> packageDependency) {
+        String dependencyPackageId = convertRawEntryToPackageId(packageDependency.getKey(), packageDependency.getValue(), linkedPackageResolver, reportingProjectPackagePath);
+        Optional<Dependency> child = buildDependencyFromPackageId(dependencyPackageId);
+        child.ifPresent(c -> graphBuilder.addChildWithParent(child.get(), pnpmPackage.get()));
+    }
+
+    private PnpmProjectPackagev6 convertPnpmLockYamlToPnpmProjectPackage(PnpmLockYamlv6 pnpmLockYaml) {
+        PnpmProjectPackagev6 pnpmProjectPackage = new PnpmProjectPackagev6();
+
+        pnpmProjectPackage.dependencies = pnpmLockYaml.dependencies;
+        pnpmProjectPackage.devDependencies = pnpmLockYaml.devDependencies;
+        pnpmProjectPackage.optionalDependencies = pnpmLockYaml.optionalDependencies;
+
+        return pnpmProjectPackage;
+    }
+
+    private List<String> extractRootPackageIds(
+        PnpmProjectPackagev6 pnpmProjectPackage,
+        @Nullable String reportingProjectPackagePath,
+        PnpmLinkedPackageResolver linkedPackageResolver
+    ) {
+        Map<String, PnpmDependencyInfo> rawPackageInfo = new HashMap<>();
+        if (pnpmProjectPackage.dependencies != null) {
+            rawPackageInfo.putAll(pnpmProjectPackage.dependencies);
+        }
+        dependencyTypeFilter.ifShouldInclude(PnpmDependencyType.DEV, pnpmProjectPackage.devDependencies, rawPackageInfo::putAll);
+        dependencyTypeFilter.ifShouldInclude(PnpmDependencyType.OPTIONAL, pnpmProjectPackage.optionalDependencies, rawPackageInfo::putAll);
+
+        return rawPackageInfo.entrySet().stream()
+            .map(entry -> convertRawEntryToPackageId(entry.getKey(), entry.getValue().version, linkedPackageResolver, reportingProjectPackagePath))
+            .collect(Collectors.toList());
+    }
+
+    private String convertRawEntryToPackageId(String name, String version, PnpmLinkedPackageResolver linkedPackageResolver, @Nullable String reportingProjectPackagePath) {
+        name = StringUtils.strip(name, "'");
+        if (version.startsWith(LINKED_PACKAGE_PREFIX)) {
+            // a linked project package's version will be referenced in the format: <linkPrefix><pathToLinkedPackageRelativeToReportingProjectPackage>
+            version = linkedPackageResolver.resolveVersionOfLinkedPackage(reportingProjectPackagePath, version.replace(LINKED_PACKAGE_PREFIX, ""));
+        }
+        return String.format("/%s@%s", name, version);
```

Comment:
A lot of v6 parsing, primarily in the packages area, use a @ as a separator between package name and version. When we construct our own Strings we should therefore use @ as well so our matching code works.